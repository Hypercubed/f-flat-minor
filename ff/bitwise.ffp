.load ./lib/core.ff

0 1 0 1 . clr
0 0 1 1 . clr

0 '&' 10 println

0 0 &
1 0 &
0 1 &
1 1 &

. clr

0 '|' 10 println

0 0 |
1 0 |
0 1 |
1 1 |

. clr

0 '~' 10 println

0 ~
1 ~

. clr


0 '5\s3\s&' 10 println

5 3 & . clr

0 '5\s3\s|' 10 println

5 3 | . clr

0 '5\s3\sbxor' 10 println

5 3 bxor . clr


