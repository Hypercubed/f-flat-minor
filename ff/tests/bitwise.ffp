.load ../lib/core.ff

0 1 0 1 . clr
0 0 1 1 . clr

0 10 '&\sTable' println

0 0 &
1 0 &
0 1 &
1 1 &

. clr

0 10 '|\sTable' println

0 0 |
1 0 |
0 1 |
1 1 |

. clr

0 10 '~\sTable' println

0 ~
1 ~
0 ~
1 ~

. clr


0 10 '5\s3\s&' println

5 3 & . clr

0 10 '5\s3\s|' println

5 3 | . clr

0 10 '5\s3\sbxor' println

5 3 bxor . clr


