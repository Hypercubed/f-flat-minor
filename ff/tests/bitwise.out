[ 0 1 0 1 ]
[ 0 0 1 1 ]

&
[ 0 0 0 1 ]

|
[ 0 1 1 1 ]

~
[ -1 -2 ]

5 3 &
[ 1 ]

5 3 |
[ 7 ]

5 3 bxor
[ 6 ]
